src/analysis/processing/qgsalgorithmrasterrank.cpp

```diff
@@ -0,0 +1,309 @@
+  {
+    outputExtent = parameterAsExtent( parameters, QStringLiteral( "EXTENT" ), context, outputCrs );
+  }
+  else
+  {
+    outputExtent = QgsProcessingUtils::combineLayerExtents( mLayers, outputCrs, context );
+  }
+
+  double minCellSizeX = 1e9;
+  double minCellSizeY = 1e9;
+  for ( QgsMapLayer *layer : mLayers )
+  {
+    QgsRasterLayer *rLayer = qobject_cast<QgsRasterLayer *>( layer );
+
+    QgsRectangle extent = rLayer->extent();
+    if ( rLayer->crs() != outputCrs )
+    {
+      QgsCoordinateTransform ct( rLayer->crs(), outputCrs, context.transformContext() );
+      extent = ct.transformBoundingBox( extent );
+    }
+
+    minCellSizeX = std::min( minCellSizeX, ( extent.xMaximum() - extent.xMinimum() ) / rLayer->width() );
+    minCellSizeY = std::min( minCellSizeY, ( extent.yMaximum() - extent.yMinimum() ) / rLayer->height() );
+  }
+
+  double outputCellSizeX = parameterAsDouble( parameters, QStringLiteral( "CELL_SIZE" ), context );
+  double outputCellSizeY = outputCellSizeX;
+  if ( outputCellSizeX == 0 )
+  {
+    outputCellSizeX = minCellSizeX;
+    outputCellSizeY = minCellSizeY;
+  }
+
+  double cols = std::round( ( outputExtent.xMaximum() - outputExtent.xMinimum() ) / outputCellSizeX );
+  double rows = std::round( ( outputExtent.yMaximum() - outputExtent.yMinimum() ) / outputCellSizeY );
+
+  const QString outputFile = parameterAsOutputLayer( parameters, QStringLiteral( "OUTPUT" ), context );
+  const QFileInfo fi( outputFile );
+  const QString outputFormat = QgsRasterFileWriter::driverForExtension( fi.suffix() );
+
+  auto writer = std::make_unique<QgsRasterFileWriter>( outputFile );
+  writer->setOutputFormat( outputFormat );
+  std::unique_ptr<QgsRasterDataProvider> provider( writer->createMultiBandRaster( outputDataType, cols, rows, outputExtent, outputCrs, mRanks.size() ) );
+  if ( !provider )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output: %1" ).arg( outputFile ) );
+  if ( !provider->isValid() )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output %1: %2" ).arg( outputFile, provider->error().message( QgsErrorMessage::Text ) ) );
+  provider->setNoDataValue( 1, outputNoData );
+
+  std::map<QString, std::unique_ptr<QgsRasterInterface>> inputInterfaces;
+  std::map<QString, std::unique_ptr<QgsRasterBlock>> inputBlocks;
+  for ( QgsMapLayer *layer : mLayers )
+  {
+    QgsRasterLayer *rLayer = qobject_cast<QgsRasterLayer *>( layer );
+    if ( rLayer->crs() != outputCrs )
+    {
+      QgsRasterProjector *projector = new QgsRasterProjector();
+      projector->setCrs( rLayer->crs(), outputCrs, context.transformContext() );
+      projector->setInput( rLayer->dataProvider() );
+      projector->setPrecision( QgsRasterProjector::Exact );
+      inputInterfaces[rLayer->id()].reset( projector );
+    }
+    else
+    {
+      inputInterfaces[rLayer->id()].reset( rLayer->dataProvider()->clone() );
```

Comment:
Let's avoid the unnecessary provider clone:
```suggestion
      inputInterfaces[rLayer->id()].reset( rLayer->dataProvider() );
```